### PR TITLE
Reorganize into multiple projects

### PR DESCRIPTION
**Pull request by wemrysi**
Splits the `scalacheck.Arbitrary` instances and the tests into their own sub-projects. Makes it easier to depend on the instances in other projects for tests.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -99,18 +99,61 @@ lazy val publishSettings = Seq(
...
+
+lazy val noPublishSettings = Seq(
+  publish := (),
+  publishLocal := (),
+  publishArtifact := false
+)
 
 lazy val root = (project in file("."))
+  .aggregate(core, scalacheck, tests)
+  .settings(noPublishSettings)
```
Comment (jedesah) [inline build.sbt]: Is this preferable to publishing all of the sub-projects?

**Comment by jedesah** [comment]
Curious about the above but aside from that LGTM. Btw, I added this in the PathyGen in quasar recently:

https://github.com/jedesah/Quasar/commit/4331d65c1ff85f24c1bd75264822c9b5a8f3b2e4#diff-40e2ae56540e33fc9d065079bbe40c0f

Since you are publishing a new version, might be worth putting that in here if it's deemed generic enough. I think it's probably fairly generic especially if we make it more like a function that accepts an `Arbitrary[String]` as the segment generator.

File: build.sbt (hunk 1 above)
Comment (wemrysi) [inline build.sbt]: We are publishing all the sub-projects. My understanding is that this prevents an empty `pathy` jar from being created as now there will be `pathy-core` and `pathy-scalacheck`. Did I misunderstand your question?

**Comment by wemrysi** [comment]
I'd say we do a separate PR for the other change you referenced. Once you're happy with this PR I'm going to publish another release so we can depend on it in quasar. There's bound to be a bit of release churn while we sort this stuff out, which is also why we're still in `0.0.x`.

File: build.sbt (hunk 1 above)
Comment (jedesah) [inline build.sbt]: so if you run `publish` in root, what happens, do all the sub-projects get published or does nothing get published?

Comment (wemrysi) [inline build.sbt]: All the subprojects are published.

**Comment by jedesah** [comment]
It doesn't need to be in this PR, but it might make sense to open a quick new PR before you publish. My reasoning is that one of the main intents of this PR is to be able to use the Generators in Quasar and get rid of `PathyGen` in Quasar. However, without this we won't be able to delete `PathyGen`.

File: build.sbt (hunk 1 above)
Comment (jedesah) [inline build.sbt]: cool! that sounds good to me. Was not obvious from looking at it though...

**Pull request merged**